addons/api/tests/unit/serializers/host-catalog-test.js

```diff
@@ -0,0 +1,187 @@
+      type: 'static',
+    };
+    assert.deepEqual(record.serialize(), expectedResult);
+  });
+
+  test('it serializes an AWS plugin as expected, ignoring Azure fields', async function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const record = store.createRecord('host-catalog', {
+      compositeType: 'aws',
+      name: 'AWS',
+      description: 'this is a Aws plugin host-catalog',
+      disable_credential_rotation: true,
+      // these are AWS fields and should be included
+      region: 'spain',
+      access_key_id: 'foobars',
+      secret_access_key: 'testing',
+      // these are Azure fields and should be excluded
+      tenant_id: 'a1b2c3',
+      client_id: 'a1b2c3',
+      subscription_id: 'a1b2c3',
+      secret_id: 'a1b2c3',
+      secret_value: 'a1b2c3',
+    });
+    const expectedResult = {
+      name: 'AWS',
+      description: 'this is a Aws plugin host-catalog',
+      type: 'plugin',
+      attributes: {
+        disable_credential_rotation: true,
+        region: 'spain',
+      },
+      secrets: {
+        access_key_id: 'foobars',
+        secret_access_key: 'testing',
+      },
+    };
+    assert.deepEqual(record.serialize(), expectedResult);
+  });
+
+  test('it serializes an Azure plugin as expected, ignoring AWS fields', async function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const record = store.createRecord('host-catalog', {
+      compositeType: 'azure',
+      name: 'Azure',
+      description: 'this is a Azure plugin host-catalog',
+      disable_credential_rotation: true,
+      // these are AWS fields and should be included
+      region: 'spain',
+      access_key_id: 'foobars',
+      secret_access_key: 'testing',
+      // these are Azure fields and should be excluded
+      tenant_id: 'a1b2c3',
+      client_id: 'a1b2c3',
+      subscription_id: 'a1b2c3',
+      secret_id: 'a1b2c3',
+      secret_value: 'a1b2c3',
+    });
+    const expectedResult = {
+      name: 'Azure',
+      description: 'this is a Azure plugin host-catalog',
+      type: 'plugin',
+      attributes: {
+        disable_credential_rotation: true,
+        tenant_id: 'a1b2c3',
+        client_id: 'a1b2c3',
+        subscription_id: 'a1b2c3',
+      },
+      secrets: {
+        secret_id: 'a1b2c3',
+        secret_value: 'a1b2c3',
+      },
+    };
+    assert.deepEqual(record.serialize(), expectedResult);
+  });
+
+  test('it serializes aws correctly on update', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    store.push({
+      data: {
+        id: '1',
+        type: 'host-catalog',
+        attributes: {
+          type: 'plugin',
+          name: 'aws',
+          description: 'test description',
+          plugin: {
+            name: 'aws',
+          },
+          region: 'andorra',
+          access_key_id: 'foobars',
```

Comment:
With `store.push` we're simulating a "saved" record.  Saved records would not have values for secrets fields.  Your test case is _technically_ correct but this condition is not expected to occur in real life.